core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -16,26 +16,6118 @@
+
+
+class PatchedDatetimeType(type):
+    """Validates the datetime instances."""
+    def __instancecheck__(cls, other):
+        """Validates whether the given instance is a datatime
+        instance.
+        """
+        return isinstance(other, OriginalDatetimeType)
+
+
+class MockDatetime13Hours(datetime.datetime):
+    __metaclass__ = PatchedDatetimeType
+
+    @classmethod
+    def utcnow(cls):
+        """Returns the current date and time 13 hours behind of UTC."""
+        return CURRENT_DATETIME - datetime.timedelta(hours=13)
+
+
+def run_job_and_check_output(
+        self, expected_output, sort=False, literal_eval=False):
+    """Helper function to run job and compare output."""
+    job_id = self.job_class.create_new()
+    self.assertEqual(
+        self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 0)
+    self.job_class.enqueue(job_id)
+    self.assertEqual(
+        self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+    self.process_and_flush_pending_tasks()
+    actual_output = self.job_class.get_output(job_id)
+    if literal_eval:
+        actual_output_dict = {}
+        expected_output_dict = {}
+
+        for item in [ast.literal_eval(value) for value in actual_output]:
+            value = item[1]
+            if isinstance(value, list):
+                value = sorted(value)
+            actual_output_dict[item[0]] = value
+
+        for item in [ast.literal_eval(value) for value in expected_output]:
+            value = item[1]
+            if isinstance(value, list):
+                value = sorted(value)
+            expected_output_dict[item[0]] = value
+
+        self.assertEqual(
+            sorted(actual_output_dict.keys()),
+            sorted(expected_output_dict.keys()))
+        for key in actual_output_dict:
+            self.assertEqual(actual_output_dict[key], expected_output_dict[key])
+    elif sort:
+        self.assertEqual(sorted(actual_output), sorted(expected_output))
+    else:
+        self.assertEqual(actual_output, expected_output)
+
+
+def update_datastore_types_for_mock_datetime():
+    """Updates datastore types for MockDatetime13Hours to ensure that validation
+    of ndb datetime properties does not fail.
+    """
+
+    # pylint: disable=protected-access
+    datastore_types._VALIDATE_PROPERTY_VALUES[MockDatetime13Hours] = (
+        datastore_types.ValidatePropertyNothing)
+    datastore_types._PACK_PROPERTY_VALUES[MockDatetime13Hours] = (
+        datastore_types.PackDatetime)
+    datastore_types._PROPERTY_MEANINGS[MockDatetime13Hours] = (
+        datastore_types.entity_pb.Property.GD_WHEN)
+    # pylint: enable=protected-access
+
+
+class ActivityReferencesModelValidatorTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(ActivityReferencesModelValidatorTests, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(USER_EMAIL, USER_NAME)
```

Comment:
This is not used in the test - you can remove it.

Reply:
Done.